### PR TITLE
Detect outsideclick and ESC pressed when showing modal

### PR DESCRIPTION
**Pull request by timilsinabishal**

### CODE REVIEW

File: components/View/Modal/index.js [hunk 1]
```diff
@@ -56,7 +88,10 @@ export default class Modal extends React.PureComponent {
         return (
             <Portal>
                 <FocusTrap>
-                    <div className={this.getClassName()}>
+                    <div
+                        className={this.getClassName()}
+                        ref={this.onReference}
```
Comment (tnagorra) [inline components/View/Modal/index.js]: To get a simple reference, we prefer to use the new method: ``` this.myRef = React.createRef(); ```

File: components/View/Modal/index.js [hunk 2]
```diff
@@ -5,16 +5,24 @@ import React from 'react';
...
+const noop = () => {};
+
 const propTypes = {
     children: PropTypes.oneOfType([
         PropTypes.arrayOf(PropTypes.element),
         PropTypes.element,
     ]).isRequired,
     className: PropTypes.string,
+    detectESC: PropTypes.bool,
+    onOutsideClicked: PropTypes.func,
```
Comment (tnagorra) [inline components/View/Modal/index.js]: The prop name should be ```onOutsideClick```

Comment (timilsinabishal) [reply components/View/Modal/index.js]: changed the prop name

File: components/View/Modal/index.js (hunk 1 above)
Comment (timilsinabishal) [reply components/View/Modal/index.js]: @tnagorra changed the reference method

**Review by frozenhelium — CHANGES_REQUESTED** [review]
Instead of props `detectESC` and `onOutsideClick`, it'd be better to implement `onClose` function and use props `closeOnEscape` and `closeOnOutsideClick`.

Also send the reason for modal close on `onClose` function like `onClose({ escape: true })` and `onClose({ outsideClick: true })`

**Comment by timilsinabishal** [comment]
@frozenhelium made the changes to the props name and passed the reason for close as arguments.

**Pull request closed without merging**